metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py

```diff
@@ -126,26 +139,67 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         config = DeltaLakeSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    inner_data_type = ""
+
+    def _inner_field_details(self, raw_field):
+        if raw_field.get("type") == "array":
+            if isinstance(raw_field.get("elementType"), dict):
+                self.inner_data_type = (
+                    self.inner_data_type
+                    + str(raw_field.get("elementType").get("type"))
+                    + "<"
+                )
+                self._inner_field_details(raw_field.get("elementType"))
+            else:
+                self.inner_data_type = self.inner_data_type + str(
+                    raw_field.get("elementType")
+                )
+        elif raw_field.get("type") == "struct":
+            data_type = ""
+            data_type = data_type + ",".join(
+                "{0}:{1}".format(field.get("name"), field.get("type"))
+                for field in raw_field.get("fields")
+            )
+            self.inner_data_type = self.inner_data_type + data_type
+        else:
+            self.inner_data_type = self.inner_data_type + raw_field.get("elementType")
+
+    def _parse_datatype(self, raw_field_json_str: str) -> List[SchemaFieldClass]:
+        raw_field_json = json.loads(raw_field_json_str)
+
+        # get the parent field name and type
+        field_name, field_type = _get_parent_field_details(raw_field_json)
+
+        inner_field = raw_field_json.get("type")
+
+        if isinstance(inner_field, str):
+            return get_schema_fields_for_hive_column(field_name, field_type)
+        else:
+            # resetting inner_data_type
+            self.inner_data_type = ""
+
+            # recursively parse the fields to get the native data type
+            self._inner_field_details(inner_field)
+
+            for count in range(self.inner_data_type.count("<")):
+                self.inner_data_type = self.inner_data_type + ">"
```

Comment:
As discussed, I suspect this may not work for types such as `struct<field1: strcut<a:string, b:long>, field2: int>` . Please add it to your test cases and update the logic as necessary.

Reply:
@mayurinehate incorporated the comments